scripts/zones/Eastern_Altepa_Desert/mobs/Centurio_XII-I.lua

```diff
@@ -2,11 +2,12 @@
 -- Area: Eastern Altepa Desert (114)
 --   NM: Centurio XII-I
 -----------------------------------
+mixins = { require('scripts/mixins/job_special'), require('scripts/mixins/rotz_bodyguarded_nm') }
+-----------------------------------
 ---@type TMobEntity
 local entity = {}
 
-entity.onMobDeath = function(mob, player, optParams)
-end
+-- all damage taken mods and body guard functionality in the rotz_bodyguarded_nm mixin
```

Comment:
As I said in the comment above, do place the modifiers in the mob script and leave the mixing to focus solely on the bodyguard logic.